config/configtls/configtls.go

```diff
@@ -327,24 +330,25 @@ func (c TLSClientSetting) LoadTLSConfig() (*tls.Config, error) {
 }
 
 // LoadTLSConfig loads the TLS configuration.
-func (c TLSServerSetting) LoadTLSConfig() (*tls.Config, error) {
+func (c *TLSServerSetting) LoadTLSConfig() (*tls.Config, error) {
```

Comment:
Since for the same `TLSServerSetting` we may call this multiple times, it may be better to actually return a `done` callback instead and have the reloader shutdown there.

Reply:
Great catch!

Reply:
So `done` would be a function that shuts down the reloader, and is called by the user on shutdown or when they no longer need the current TLS config, is that correct?

Reply:
I believe I've implemented what was requested here 👍

Reply:
Looking more at consumers of `TLSServerSetting.LoadTLSConfig`, it looks like there isn't a simple way to keep track of the lifecycle of these objects in a uniform way externally. It may make sense to keep track internally and then call all of the shutdown methods on `TLSServerSetting.Shutdown`. This will make it so this is no longer a breaking change, and the only extra requirement on users is calling the shutdown method here.

Reply:
I implemented my proposed solution and added tests.